pkg/handler/internal/default_pruning_test.go

```diff
@@ -0,0 +1,91 @@
+/*
+Copyright 2020 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package internal_test
+
+import (
+	"encoding/json"
+	"reflect"
+	"testing"
+
+	"k8s.io/kube-openapi/pkg/handler/internal"
+
+	"github.com/go-openapi/spec"
+)
+
+func TestDefaultPruning(t *testing.T) {
```

Comment:
not sure about the necessary effort: one could 
1. fuzz a definition, call your pruning func, marshal
2. call marshal on the original definition and use string replacement to remove defaults
3. compare

Reply:
Also with some clever logic you can check that your pruning func does not change things in-place.

Reply:
Yeah, I'm happy with that level of testing, especially considering that the timeline is a little short.

Reply:
Can you at least add an already-correct schema and verify that it isn't changed further?

Reply:
Yeah, that's a good point!